### ISSUE TITLE
Building J2V8 from source?

### ISSUE THREAD

**Issue by GriffinSchneider** [eclipsesource/J2V8#79]
Hi,

I'm trying to build J2V8 (specifically the Android JAR) so I can try to address some issues we're seeing on less-common processor architectures, but I can't figure out the build process. I see that there's a Dockerfile in `v8build/`, and I've gotten that to run that successfully (I think), but I'm not sure where to go from here. Can anyone provide some assistance on how to build J2V8? If I can get my build working, I'd be happy to document the process so this question doesn't some up again :smile:

Thanks!

**Comment by djMax** [comment]
++++++++ - we really need this and are happy to contribute resources to make it better for everybody.

**Comment by KishanV** [comment]
i also want to know.

**Comment by land007** [comment]
来吧，跟大家分享一下，怎么编译的。我需要的是linux版本编译过程。

**Comment by KishanV** [comment]
i have done it in window 10.

**Comment by djMax** [comment]
Can you tell us how? :)

**Comment by KishanV** [comment]
Use this Version  of  v8 (v8-3.26.31) and build with visual studio 2013...

you will get dll files and ......

link to  ( j2v8.vcxproj ) this project. 

you can find this file from link below.
https://github.com/eclipsesource/J2V8/tree/master/jni

**Comment by ccantill** [comment]
Anyone manage to compile from source on linux? There doesn't appear to be a build script and whatever .so I get from the build.sh is just 215k, which doesn't seem right.
I need a version that works with glibc2.12 (customer runs RHEL6 exclusively) and I'm in the dark here. I'm using gcc (no clang on RHEL6).

**Comment by ccantill** [comment]
Ok, got it working finally. What I finally ended up doing was:

    clone v8 as described in the Google V8 documentation
    $ git checkout 3.26.33
    $ export CFLAGS="-fPIC"
    $ export CXXFLAGS="-fPIC -fno-strict-aliasing"
    $ make dependencies
    $ make x64.release -j4 i18nsupport=off component=static_library -j8
    cd into the jni directory
    $ g++ -fPIC -I../../v8/include -I/usr/java/jdk1.8.0_60/include \
      -I/usr/java/jdk1.8.0_60/include/linux com_eclipsesource_v8_V8Impl.cpp \
      -o libj2v8_linux_x86_64.so \
      ../../v8/out/x64.release/obj.target/tools/gyp/libv8_base.x64.a \
      ../../v8/out/x64.release/obj.target/tools/gyp/libv8_snapshot.a -lrt -lpthread -shared

**Comment by irbull** [comment]
Sorry, I missed this earlier. Here is what I use on the build server (looks
similar).

g++ -I /data/jenkins/v8_3_26/include/ -I
/data/jenkins/tools/hudson.model.JDK/jdk-7/include/ -I
/data/jenkins/tools/hudson.model.JDK/jdk-7/include/linux
 com_eclipsesource_v8_V8Impl.cpp -fPIC -shared -o libj2v8_linux_x86_64.so
-Wl,--start-group
/data/jenkins/v8_3_26/out/native/obj.target/tools/gyp/libv8_base.x64.a
/data/jenkins/v8_3_26/out/native/obj.target/tools/gyp/libv8_nosnapshot.x64.a
-Wl,--end-group -lrt

in this case, v8_3_26 is built and sitting in /data/jenkins/v8_3_26
And we are using JDK 7 to link the JNI stuff against


On Tue, Oct 6, 2015 at 9:08 AM, Christoph Cantillon <
notifications@github.com> wrote:

> Ok, got it working finally. What I finally ended up doing was:
>
> clone v8 as described in the Google V8 documentation
> $ git checkout 3.26.33
> $ export CFLAGS="-fPIC"
> $ export CXXFLAGS="-fPIC -fno-strict-aliasing"
> $ make dependencies
> $ make x64.release -j4 i18nsupport=off component=static_library -j8
> cd into the jni directory
> $ g++ -fPIC -I../../v8/include -I/usr/java/jdk1.8.0_60/include \
>   -I/usr/java/jdk1.8.0_60/include/linux com_eclipsesource_v8_V8Impl.cpp \
>   -o libj2v8_linux_x86_64.so \
>   ../../v8/out/x64.release/obj.target/tools/gyp/libv8_base.x64.a \
>   ../../v8/out/x64.release/obj.target/tools/gyp/libv8_snapshot.a -lrt -lpthread -shared
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipsesource/J2V8/issues/79#issuecomment-145913367>.
>



-- 
R. Ian Bull | EclipseSource Victoria | +1 250 477 7484
http://eclipsesource.com | http://twitter.com/eclipsesource

**Comment by land007** [comment]
Compiled, thank the two upper floors.

**Comment by land007** [comment]
Dockerfile
FROM ubuntu:14.04
MAINTAINER EclipseSource <land007@21cn.com>

RUN mv /etc/apt/sources.list /etc/apt/sources.list.bak

RUN echo "deb http://mirrors.aliyun.com/ubuntu/ trusty main restricted universe multiverse\ndeb http://mirrors.aliyun.com/ubuntu/ trusty-security main restricted universe multiverse\ndeb http://mirrors.aliyun.com/ubuntu/ trusty-updates main restricted universe multiverse\ndeb http://mirrors.aliyun.com/ubuntu/ trusty-proposed main restricted universe multiverse\ndeb http://mirrors.aliyun.com/ubuntu/ trusty-backports main restricted universe multiverse\ndeb-src http://mirrors.aliyun.com/ubuntu/ trusty main restricted universe multiverse\ndeb-src http://mirrors.aliyun.com/ubuntu/ trusty-security main restricted universe multiverse\ndeb-src http://mirrors.aliyun.com/ubuntu/ trusty-updates main restricted universe multiverse\ndeb-src http://mirrors.aliyun.com/ubuntu/ trusty-proposed main restricted universe multiverse\ndeb-src http://mirrors.aliyun.com/ubuntu/ trusty-backports main restricted universe multiverse" > /etc/apt/sources.list

# update and upgrade
RUN apt-get update && apt-get upgrade -y && apt-get clean

# install standard toolset
RUN apt-get install -q -y git subversion make gcc bzip2 python g++ libc6-dev-i386 g++-multilib openjdk-7-jdk

# ADD http://dl.google.com/android/ndk/android-ndk-r10d-linux-x86_64.bin ./
#ADD android-ndk-r10d-linux-x86_64.bin ./
#RUN chmod a+x android-ndk-r10d-linux-x86_64.bin
#RUN ./android-ndk-r10d-linux-x86_64.bin
#ENV ANDROID_NDK_ROOT /android-ndk-r10d

# Checkout
RUN mkdir /data
#RUN cd /data && svn checkout http://v8.googlecode.com/svn/branches/3.26 v8_3_26
ADD 3.26 /data/v8_3_26
ENV CCFLAGS -fPIC
ENV CXXFLAGS -fPIC
#ENV CXXFLAGS -fPIC -fno-strict-aliasing

# Build
RUN cd /data/v8_3_26 && make dependencies

RUN cd /data/v8_3_26 && make x64.release -j4 i18nsupport=off component=static_library -j8
#RUN cd /data/v8_3_26 && make native i18nsupport=off
#RUN cd /data/v8_3_26 && make android_arm.release i18nsupport=off
#RUN cd /data/v8_3_26 && make android_ia32.release i18nsupport=off

ADD jdk-8u5-linux-x64.tar.gz /data/
#RUN tar -zxvf jdk-8u5-linux-x64.tar.gz

RUN cd /data/ && git clone https://github.com/eclipsesource/J2V8.git

RUN cd /data/J2V8/jni && g++ -fPIC -I../../v8_3_26/include -I/data/jdk1.8.0_05/include -I/data/jdk1.8.0_05/include/linux com_eclipsesource_v8_V8Impl.cpp   -o libj2v8_linux_x86_64.so   ../../v8_3_26/out/x64.release/obj.target/tools/gyp/libv8_base.x64.a   ../../v8_3_26/out/x64.release/obj.target/tools/gyp/libv8_snapshot.a -lrt -lpthread -shared

**Comment by land007** [comment]
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fd3610f6874, pid=8, tid=140545864185600
#
# JRE version: Java(TM) SE Runtime Environment (7.0_67-b01) (build 1.7.0_67-b01)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (24.65-b04 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.so+0x64d874]  get_method_id(JNIEnv_*, _jclass*, char const*, char const*, bool, Thread*)+0x84
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /data/java/hs_err_pid8.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
#
Aborted

**Comment by land007** [comment]
#
	# A fatal error has been detected by the Java Runtime Environment:
	#
	#  SIGSEGV (0xb) at pc=0x00007f27c1da7874, pid=28, tid=139808753940224
	#
	# JRE version: Java(TM) SE Runtime Environment (7.0_67-b01) (build 1.7.0_67-b01)
	# Java VM: Java HotSpot(TM) 64-Bit Server VM (24.65-b04 mixed mode linux-amd64 compressed oops)
	# Problematic frame:
	# V  [libjvm.so+0x64d874]  get_method_id(JNIEnv_*, _jclass*, char const*, char const*, bool, Thread*)+0x84
	#
	# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
	#
	# If you would like to submit a bug report, please visit:
	#   http://bugreport.sun.com/bugreport/crash.jsp
	#
	
	---------------  T H R E A D  ---------------
	
	Current thread (0x00007f27bc009800):  JavaThread "main" [_thread_in_vm, id=29, stack(0x00007f27c30d7000,0x00007f27c31d8000)]
	
	siginfo:si_signo=SIGSEGV: si_errno=0, si_code=1 (SEGV_MAPERR), si_addr=0x0000000000000000
	
	Registers:
	RAX=0x0000000000000000, RBX=0x00007f27c31d63ac, RCX=0x0000000000000000, RDX=0x0000000000000028
	RSP=0x00007f27c31d6380, RBP=0x00007f27c31d63e0, RSI=0x00007f27b4a8849b, RDI=0x00007f27b681df89
	R8 =0x0000000000000000, R9 =0x00007f27bc009800, R10=0x72732f746f707374, R11=0x000000000000007b
	R12=0x00007f27b4a884a0, R13=0x00007f27b681df80, R14=0x00007f27b68527d0, R15=0x00007f27bc009800
	RIP=0x00007f27c1da7874, EFLAGS=0x0000000000010202, CSGSFS=0x0000000000000033, ERR=0x0000000000000004
	  TRAPNO=0x000000000000000e
	
	Top of Stack: (sp=0x00007f27c31d6380)
	0x00007f27c31d6380:   00007f27b453c6d0 00007f27bc0b4880
	0x00007f27c31d6390:   00007f27b4a884a0 0000000000000000
	0x00007f27c31d63a0:   00007f27c31d6470 00009b69bc006110
	0x00007f27c31d63b0:   00000000ffffffff 00007f27bc009800
	0x00007f27c31d63c0:   00007f27c31d6400 00007f27bc0099e8
	0x00007f27c31d63d0:   00007f27b4a8849c 00007f27b4a884a0
	0x00007f27c31d63e0:   00007f27c31d6460 00007f27c1da7d60
	0x00007f27c31d63f0:   00007f27b44c5000 0000000000000000
	0x00007f27c31d6400:   00007f27bc009800 00007f27bc00a458
	0x00007f27c31d6410:   00007f270000063e 00007f27c21553d8
	0x00007f27c31d6420:   00007f27bc009800 00007f27b470d2b9
	0x00007f27c31d6430:   00007f27c31d64c0 00007f27b470d2b9
	0x00007f27c31d6440:   00007f27bc0099e8 00007f27bc0a58c0
	0x00007f27c31d6450:   00007f27bc0b4880 00007f27bc009800
	0x00007f27c31d6460:   00007f27c31d6490 00007f27b4723d52
	0x00007f27c31d6470:   00007f27b4a8849c 00007f27b4a884a0
	0x00007f27c31d6480:   0000000000000000 00007f27bc0099e8
	0x00007f27c31d6490:   00007f27c31d64c0 00007f27b470d692
	0x00007f27c31d64a0:   0000000000000000 00007f27c2571048
	0x00007f27c31d64b0:   00007f27bc0099e8 ffffffffb470d2b9
	0x00007f27c31d64c0:   00007f27c31d6620 00007f27c0e2aaac
	0x00007f27c31d64d0:   00007f27c31d66a0 00007f27c31d66a8
	0x00007f27c31d64e0:   00007f27b9005310 00007f2700000000
	0x00007f27c31d64f0:   00007f27c31d6620 00007f27c31d65c0
	0x00007f27c31d6500:   00007f27c31d65c0 00007f27b901da83
	0x00007f27c31d6510:   00007f27b9005310 00007f2700000100
	0x00007f27c31d6520:   00007f27c31d6620 00007f27c31d64d8
	0x00007f27c31d6530:   00007f27c31d6519 0000000800000001
	0x00007f27c31d6540:   0000000000000000 00007f27c31d6548
	0x00007f27c31d6550:   000000000000000e 00007f2700000000
	0x00007f27c31d6560:   00000000bd2ad6a0 00007f27bc009800
	0x00007f27c31d6570:   0000000000000000 0000000000000001 
	
	Instructions: (pc=0x00007f27c1da7874)
	0x00007f27c1da7854:   89 c6 e8 15 cc 2c 00 4d 85 f6 49 89 c5 0f 84 79
	0x00007f27c1da7864:   02 00 00 48 85 c0 0f 84 10 01 00 00 48 8b 45 b8
	0x00007f27c1da7874:   48 8b 38 e8 d4 2f fb ff 84 c0 0f 85 cc 00 00 00
	0x00007f27c1da7884:   48 8b 45 b8 31 db 48 8b 38 e8 ce 2e fb ff 48 85 
	
	Register to memory mapping:
	
	RAX=0x0000000000000000 is an unknown value
	RBX=0x00007f27c31d63ac is pointing into the stack for thread: 0x00007f27bc009800
	RCX=0x0000000000000000 is an unknown value
	RDX=0x0000000000000028 is an unknown value
	RSP=0x00007f27c31d6380 is pointing into the stack for thread: 0x00007f27bc009800
	RBP=0x00007f27c31d63e0 is pointing into the stack for thread: 0x00007f27bc009800
	RSI=0x00007f27b4a8849b: <offset 0x5c349b> in /data/J2V8/jni/libj2v8_linux_x86_64.so at 0x00007f27b44c5000
	RDI=0x00007f27b681df89 is an unknown value
	R8 =0x0000000000000000 is an unknown value
	R9 =0x00007f27bc009800 is a thread
	R10=0x72732f746f707374 is an unknown value
	R11=0x000000000000007b is an unknown value
	R12=0x00007f27b4a884a0: <offset 0x5c34a0> in /data/J2V8/jni/libj2v8_linux_x86_64.so at 0x00007f27b44c5000
	R13=0x00007f27b681df80 is an unknown value
	R14=0x00007f27b68527d0 is an unknown value
	R15=0x00007f27bc009800 is a thread
	
	
	Stack: [0x00007f27c30d7000,0x00007f27c31d8000],  sp=0x00007f27c31d6380,  free space=1020k
	Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
	V  [libjvm.so+0x64d874]  get_method_id(JNIEnv_*, _jclass*, char const*, char const*, bool, Thread*)+0x84
	V  [libjvm.so+0x64dd60]  jni_GetMethodID+0xb0
	C  [libj2v8_linux_x86_64.so+0x25ed52]  JNIEnv_::GetMethodID(_jclass*, char const*, char const*)+0x3c
	C  [libj2v8_linux_x86_64.so+0x248692]  JNI_OnLoad+0x3d9
	C  [libjava.so+0xdaac]  Java_java_lang_ClassLoader_00024NativeLibrary_load+0x26c
	j  java.lang.ClassLoader$NativeLibrary.load(Ljava/lang/String;)V+0
	j  java.lang.ClassLoader.loadLibrary1(Ljava/lang/Class;Ljava/io/File;)Z+302
	j  java.lang.ClassLoader.loadLibrary0(Ljava/lang/Class;Ljava/io/File;)Z+2
	j  java.lang.ClassLoader.loadLibrary(Ljava/lang/Class;Ljava/lang/String;Z)V+48
	j  java.lang.Runtime.load0(Ljava/lang/Class;Ljava/lang/String;)V+57
	j  java.lang.System.load(Ljava/lang/String;)V+7
	j  test.Test1.main([Ljava/lang/String;)V+2
	v  ~StubRoutines::call_stub
	V  [libjvm.so+0x5ffb65]  JavaCalls::call_helper(JavaValue*, methodHandle*, JavaCallArguments*, Thread*)+0x365
	V  [libjvm.so+0x5fe5c8]  JavaCalls::call(JavaValue*, methodHandle, JavaCallArguments*, Thread*)+0x28
	V  [libjvm.so+0x638309]  jni_invoke_static(JNIEnv_*, JavaValue*, _jobject*, JNICallType, _jmethodID*, JNI_ArgumentPusher*, Thread*)+0x219
	V  [libjvm.so+0x640f22]  jni_CallStaticVoidMethod+0x162
	C  [libjli.so+0x36d9]  JavaMain+0x7e9
	
	Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
	j  java.lang.ClassLoader$NativeLibrary.load(Ljava/lang/String;)V+0
	j  java.lang.ClassLoader.loadLibrary1(Ljava/lang/Class;Ljava/io/File;)Z+302
	j  java.lang.ClassLoader.loadLibrary0(Ljava/lang/Class;Ljava/io/File;)Z+2
	j  java.lang.ClassLoader.loadLibrary(Ljava/lang/Class;Ljava/lang/String;Z)V+48
	j  java.lang.Runtime.load0(Ljava/lang/Class;Ljava/lang/String;)V+57
	j  java.lang.System.load(Ljava/lang/String;)V+7
	j  test.Test1.main([Ljava/lang/String;)V+2
	v  ~StubRoutines::call_stub
	
	---------------  P R O C E S S  ---------------
	
	Java Threads: ( => current thread )
	  0x00007f27bc094800 JavaThread "Service Thread" daemon [_thread_blocked, id=40, stack(0x00007f27b4f00000,0x00007f27b5001000)]
	  0x00007f27bc092800 JavaThread "C2 CompilerThread1" daemon [_thread_blocked, id=39, stack(0x00007f27b5001000,0x00007f27b5102000)]
	  0x00007f27bc08f800 JavaThread "C2 CompilerThread0" daemon [_thread_blocked, id=38, stack(0x00007f27b5102000,0x00007f27b5203000)]
	  0x00007f27bc087000 JavaThread "Signal Dispatcher" daemon [_thread_blocked, id=37, stack(0x00007f27b5203000,0x00007f27b5304000)]
	  0x00007f27bc06f800 JavaThread "Finalizer" daemon [_thread_blocked, id=36, stack(0x00007f27b5304000,0x00007f27b5405000)]
	  0x00007f27bc06d800 JavaThread "Reference Handler" daemon [_thread_blocked, id=35, stack(0x00007f27b5405000,0x00007f27b5506000)]
	=>0x00007f27bc009800 JavaThread "main" [_thread_in_vm, id=29, stack(0x00007f27c30d7000,0x00007f27c31d8000)]
	
	Other Threads:
	  0x00007f27bc069000 VMThread [stack: 0x00007f27b5506000,0x00007f27b5607000] [id=34]
	  0x00007f27bc09f000 WatcherThread [stack: 0x00007f27b4dff000,0x00007f27b4f00000] [id=41]
	
	VM state:not at safepoint (normal execution)
	
	VM Mutex/Monitor currently owned by a thread: None
	
	Heap
	 PSYoungGen      total 18944K, used 983K [0x00000000eb680000, 0x00000000ecb80000, 0x0000000100000000)
	  eden space 16384K, 6% used [0x00000000eb680000,0x00000000eb775db0,0x00000000ec680000)
	  from space 2560K, 0% used [0x00000000ec900000,0x00000000ec900000,0x00000000ecb80000)
	  to   space 2560K, 0% used [0x00000000ec680000,0x00000000ec680000,0x00000000ec900000)
	 ParOldGen       total 41984K, used 0K [0x00000000c2400000, 0x00000000c4d00000, 0x00000000eb680000)
	  object space 41984K, 0% used [0x00000000c2400000,0x00000000c2400000,0x00000000c4d00000)
	 PSPermGen       total 21504K, used 2550K [0x00000000bd200000, 0x00000000be700000, 0x00000000c2400000)
	  object space 21504K, 11% used [0x00000000bd200000,0x00000000bd47dbd0,0x00000000be700000)
	
	Card table byte_map: [0x00007f27c01ae000,0x00007f27c03c6000] byte_map_base: 0x00007f27bfbc5000
	
	Polling page: 0x00007f27c31e3000
	
	Code Cache  [0x00007f27b9000000, 0x00007f27b9270000, 0x00007f27bc000000)
	 total_blobs=167 nmethods=0 adapters=136 free_code_cache=48773Kb largest_free_block=49944064
	
	Compilation events (0 events):
	No events
	
	GC Heap History (0 events):
	No events
	
	Deoptimization events (0 events):
	No events
	
	Internal exceptions (10 events):
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb724e58 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jvm.cpp:1244
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb725368 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jvm.cpp:1244
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb725520 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/classfile/systemDictionary.cpp:174
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb725be0 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jvm.cpp:1244
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb7260e0 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jvm.cpp:1244
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb726290 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/classfile/systemDictionary.cpp:174
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb7268f0 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jvm.cpp:1244
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb726da0 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jvm.cpp:1244
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb726f40 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/classfile/systemDictionary.cpp:174
	Event: 0.052 Thread 0x00007f27bc009800 Threw 0x00000000eb727258 at /HUDSON/workspace/7u-2-build-linux-amd64/jdk7u67/1368/hotspot/src/share/vm/prims/jni.cpp:1598
	
	Events (10 events):
	Event: 0.052 loading class 0x00007f27bc0b64a0
	Event: 0.052 loading class 0x00007f27bc0b64a0 done
	Event: 0.052 loading class 0x00007f27bc0b6520
	Event: 0.052 loading class 0x00007f27bc0b6520 done
	Event: 0.052 loading class 0x00007f27bc0b65a0
	Event: 0.052 loading class 0x00007f27bc0b65a0 done
	Event: 0.052 loading class 0x00007f27bc0b65f0
	Event: 0.052 loading class 0x00007f27bc0b65f0 done
	Event: 0.052 loading class 0x00007f27bc0b6640
	Event: 0.052 loading class 0x00007f27bc0b6640 done
	
	
	Dynamic libraries:
	00400000-00401000 r-xp 00000000 00:20 89                                 /data/jdk1.7.0_67/bin/java
	00600000-00601000 rw-p 00000000 00:20 89                                 /data/jdk1.7.0_67/bin/java
	00e7e000-00e9f000 rw-p 00000000 00:00 0                                  [heap]
	bd200000-be700000 rw-p 00000000 00:00 0 
	be700000-c2400000 rw-p 00000000 00:00 0 
	c2400000-c4d00000 rw-p 00000000 00:00 0 
	c4d00000-eb680000 rw-p 00000000 00:00 0 
	eb680000-ecb80000 rw-p 00000000 00:00 0 
	ecb80000-100000000 rw-p 00000000 00:00 0 
	7f2780000000-7f2780021000 rw-p 00000000 00:00 0 
	7f2780021000-7f2784000000 ---p 00000000 00:00 0 
	7f2787dea000-7f2787dff000 r-xp 00000000 00:20 136                        /lib/x86_64-linux-gnu/libgcc_s.so.1
	7f2787dff000-7f2787ffe000 ---p 00015000 00:20 136                        /lib/x86_64-linux-gnu/libgcc_s.so.1
	7f2787ffe000-7f2787fff000 r--p 00014000 00:20 136                        /lib/x86_64-linux-gnu/libgcc_s.so.1
	7f2787fff000-7f2788000000 rw-p 00015000 00:20 136                        /lib/x86_64-linux-gnu/libgcc_s.so.1
	7f2788000000-7f2788021000 rw-p 00000000 00:00 0 
	7f2788021000-7f278c000000 ---p 00000000 00:00 0 
	7f278c000000-7f278c021000 rw-p 00000000 00:00 0 
	7f278c021000-7f2790000000 ---p 00000000 00:00 0 
	7f2790000000-7f2790021000 rw-p 00000000 00:00 0 
	7f2790021000-7f2794000000 ---p 00000000 00:00 0 
	7f2794000000-7f2794021000 rw-p 00000000 00:00 0 
	7f2794021000-7f2798000000 ---p 00000000 00:00 0 
	7f2798000000-7f2798021000 rw-p 00000000 00:00 0 
	7f2798021000-7f279c000000 ---p 00000000 00:00 0 
	7f279c000000-7f279c021000 rw-p 00000000 00:00 0 
	7f279c021000-7f27a0000000 ---p 00000000 00:00 0 
	7f27a0000000-7f27a0021000 rw-p 00000000 00:00 0 
	7f27a0021000-7f27a4000000 ---p 00000000 00:00 0 
	7f27a4000000-7f27a4021000 rw-p 00000000 00:00 0 
	7f27a4021000-7f27a8000000 ---p 00000000 00:00 0 
	7f27a8000000-7f27a8021000 rw-p 00000000 00:00 0 
	7f27a8021000-7f27ac000000 ---p 00000000 00:00 0 
	7f27ac000000-7f27ac021000 rw-p 00000000 00:00 0 
	7f27ac021000-7f27b0000000 ---p 00000000 00:00 0 
	7f27b0000000-7f27b0021000 rw-p 00000000 00:00 0 
	7f27b0021000-7f27b4000000 ---p 00000000 00:00 0 
	7f27b4051000-7f27b41bd000 rw-p 00000000 00:00 0 
	7f27b41bd000-7f27b429f000 r-xp 00000000 00:20 135                        /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.16
	7f27b429f000-7f27b449e000 ---p 000e2000 00:20 135                        /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.16
	7f27b449e000-7f27b44a6000 r--p 000e1000 00:20 135                        /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.16
	7f27b44a6000-7f27b44a8000 rw-p 000e9000 00:20 135                        /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.16
	7f27b44a8000-7f27b44bd000 rw-p 00000000 00:00 0 
	7f27b44c5000-7f27b4bb8000 r-xp 00000000 00:20 132                        /data/J2V8/jni/libj2v8_linux_x86_64.so
	7f27b4bb8000-7f27b4db7000 ---p 006f3000 00:20 132                        /data/J2V8/jni/libj2v8_linux_x86_64.so
	7f27b4db7000-7f27b4de9000 r--p 006f2000 00:20 132                        /data/J2V8/jni/libj2v8_linux_x86_64.so
	7f27b4de9000-7f27b4dfb000 rw-p 00724000 00:20 132                        /data/J2V8/jni/libj2v8_linux_x86_64.so
	7f27b4dfb000-7f27b4dff000 rw-p 00000000 00:00 0 
	7f27b4dff000-7f27b4e00000 ---p 00000000 00:00 0 
	7f27b4e00000-7f27b4f00000 rw-p 00000000 00:00 0                          [stack:41]
	7f27b4f00000-7f27b4f03000 ---p 00000000 00:00 0 
	7f27b4f03000-7f27b5001000 rw-p 00000000 00:00 0                          [stack:40]
	7f27b5001000-7f27b5004000 ---p 00000000 00:00 0 
	7f27b5004000-7f27b5102000 rw-p 00000000 00:00 0                          [stack:39]
	7f27b5102000-7f27b5105000 ---p 00000000 00:00 0 
	7f27b5105000-7f27b5203000 rw-p 00000000 00:00 0                          [stack:38]
	7f27b5203000-7f27b5206000 ---p 00000000 00:00 0 
	7f27b5206000-7f27b5304000 rw-p 00000000 00:00 0                          [stack:37]
	7f27b5304000-7f27b5307000 ---p 00000000 00:00 0 
	7f27b5307000-7f27b5405000 rw-p 00000000 00:00 0                          [stack:36]
	7f27b5405000-7f27b5408000 ---p 00000000 00:00 0 
	7f27b5408000-7f27b5506000 rw-p 00000000 00:00 0                          [stack:35]
	7f27b5506000-7f27b5507000 ---p 00000000 00:00 0 
	7f27b5507000-7f27b65e5000 rw-p 00000000 00:00 0                          [stack:34]
	7f27b65e5000-7f27b67a4000 r--s 039e5000 00:20 114                        /data/jdk1.7.0_67/jre/lib/rt.jar
	7f27b67a4000-7f27b8bfc000 rw-p 00000000 00:00 0 
	7f27b8bfc000-7f27b8bfd000 ---p 00000000 00:00 0 
	7f27b8bfd000-7f27b8cfd000 rw-p 00000000 00:00 0                          [stack:33]
	7f27b8cfd000-7f27b8cfe000 ---p 00000000 00:00 0 
	7f27b8cfe000-7f27b8dfe000 rw-p 00000000 00:00 0                          [stack:32]
	7f27b8dfe000-7f27b8dff000 ---p 00000000 00:00 0 
	7f27b8dff000-7f27b8eff000 rw-p 00000000 00:00 0                          [stack:31]
	7f27b8eff000-7f27b8f00000 ---p 00000000 00:00 0 
	7f27b8f00000-7f27b9000000 rw-p 00000000 00:00 0                          [stack:30]
	7f27b9000000-7f27b9270000 rwxp 00000000 00:00 0 
	7f27b9270000-7f27bc1a7000 rw-p 00000000 00:00 0 
	7f27bc1a7000-7f27c0000000 ---p 00000000 00:00 0 
	7f27c0006000-7f27c0079000 rw-p 00000000 00:00 0 
	7f27c0079000-7f27c01ae000 rw-p 00000000 00:00 0 
	7f27c01ae000-7f27c01b9000 rw-p 00000000 00:00 0 
	7f27c01b9000-7f27c01d7000 rw-p 00000000 00:00 0 
	7f27c01d7000-7f27c01ec000 rw-p 00000000 00:00 0 
	7f27c01ec000-7f27c0320000 rw-p 00000000 00:00 0 
	7f27c0320000-7f27c032b000 rw-p 00000000 00:00 0 
	7f27c032b000-7f27c03c5000 rw-p 00000000 00:00 0 
	7f27c03c5000-7f27c03c6000 rw-p 00000000 00:00 0 
	7f27c03c6000-7f27c03e0000 r-xp 00000000 00:20 112                        /data/jdk1.7.0_67/jre/lib/amd64/libzip.so
	7f27c03e0000-7f27c05e0000 ---p 0001a000 00:20 112                        /data/jdk1.7.0_67/jre/lib/amd64/libzip.so
	7f27c05e0000-7f27c05e1000 rw-p 0001a000 00:20 112                        /data/jdk1.7.0_67/jre/lib/amd64/libzip.so
	7f27c05e1000-7f27c05ed000 r-xp 00000000 00:20 45                         /lib/x86_64-linux-gnu/libnss_files-2.15.so
	7f27c05ed000-7f27c07ec000 ---p 0000c000 00:20 45                         /lib/x86_64-linux-gnu/libnss_files-2.15.so
	7f27c07ec000-7f27c07ed000 r--p 0000b000 00:20 45                         /lib/x86_64-linux-gnu/libnss_files-2.15.so
	7f27c07ed000-7f27c07ee000 rw-p 0000c000 00:20 45                         /lib/x86_64-linux-gnu/libnss_files-2.15.so
	7f27c07ee000-7f27c07f8000 r-xp 00000000 00:20 43                         /lib/x86_64-linux-gnu/libnss_nis-2.15.so
	7f27c07f8000-7f27c09f8000 ---p 0000a000 00:20 43                         /lib/x86_64-linux-gnu/libnss_nis-2.15.so
	7f27c09f8000-7f27c09f9000 r--p 0000a000 00:20 43                         /lib/x86_64-linux-gnu/libnss_nis-2.15.so
	7f27c09f9000-7f27c09fa000 rw-p 0000b000 00:20 43                         /lib/x86_64-linux-gnu/libnss_nis-2.15.so
	7f27c09fa000-7f27c0a11000 r-xp 00000000 00:20 41                         /lib/x86_64-linux-gnu/libnsl-2.15.so
	7f27c0a11000-7f27c0c10000 ---p 00017000 00:20 41                         /lib/x86_64-linux-gnu/libnsl-2.15.so
	7f27c0c10000-7f27c0c11000 r--p 00016000 00:20 41                         /lib/x86_64-linux-gnu/libnsl-2.15.so
	7f27c0c11000-7f27c0c12000 rw-p 00017000 00:20 41                         /lib/x86_64-linux-gnu/libnsl-2.15.so
	7f27c0c12000-7f27c0c14000 rw-p 00000000 00:00 0 
	7f27c0c14000-7f27c0c1c000 r-xp 00000000 00:20 39                         /lib/x86_64-linux-gnu/libnss_compat-2.15.so
	7f27c0c1c000-7f27c0e1b000 ---p 00008000 00:20 39                         /lib/x86_64-linux-gnu/libnss_compat-2.15.so
	7f27c0e1b000-7f27c0e1c000 r--p 00007000 00:20 39                         /lib/x86_64-linux-gnu/libnss_compat-2.15.so
	7f27c0e1c000-7f27c0e1d000 rw-p 00008000 00:20 39                         /lib/x86_64-linux-gnu/libnss_compat-2.15.so
	7f27c0e1d000-7f27c0e46000 r-xp 00000000 00:20 99                         /data/jdk1.7.0_67/jre/lib/amd64/libjava.so
	7f27c0e46000-7f27c1046000 ---p 00029000 00:20 99                         /data/jdk1.7.0_67/jre/lib/amd64/libjava.so
	7f27c1046000-7f27c1048000 rw-p 00029000 00:20 99                         /data/jdk1.7.0_67/jre/lib/amd64/libjava.so
	7f27c1048000-7f27c1055000 r-xp 00000000 00:20 107                        /data/jdk1.7.0_67/jre/lib/amd64/libverify.so
	7f27c1055000-7f27c1254000 ---p 0000d000 00:20 107                        /data/jdk1.7.0_67/jre/lib/amd64/libverify.so
	7f27c1254000-7f27c1256000 rw-p 0000c000 00:20 107                        /data/jdk1.7.0_67/jre/lib/amd64/libverify.so
	7f27c1256000-7f27c125d000 r-xp 00000000 00:20 106                        /lib/x86_64-linux-gnu/librt-2.15.so
	7f27c125d000-7f27c145c000 ---p 00007000 00:20 106                        /lib/x86_64-linux-gnu/librt-2.15.so
	7f27c145c000-7f27c145d000 r--p 00006000 00:20 106                        /lib/x86_64-linux-gnu/librt-2.15.so
	7f27c145d000-7f27c145e000 rw-p 00007000 00:20 106                        /lib/x86_64-linux-gnu/librt-2.15.so
	7f27c145e000-7f27c1559000 r-xp 00000000 00:20 104                        /lib/x86_64-linux-gnu/libm-2.15.so
	7f27c1559000-7f27c1758000 ---p 000fb000 00:20 104                        /lib/x86_64-linux-gnu/libm-2.15.so
	7f27c1758000-7f27c1759000 r--p 000fa000 00:20 104                        /lib/x86_64-linux-gnu/libm-2.15.so
	7f27c1759000-7f27c175a000 rw-p 000fb000 00:20 104                        /lib/x86_64-linux-gnu/libm-2.15.so
	7f27c175a000-7f27c22ca000 r-xp 00000000 00:20 102                        /data/jdk1.7.0_67/jre/lib/amd64/server/libjvm.so
	7f27c22ca000-7f27c24c9000 ---p 00b70000 00:20 102                        /data/jdk1.7.0_67/jre/lib/amd64/server/libjvm.so
	7f27c24c9000-7f27c258e000 rw-p 00b6f000 00:20 102                        /data/jdk1.7.0_67/jre/lib/amd64/server/libjvm.so
	7f27c258e000-7f27c25ce000 rw-p 00000000 00:00 0 
	7f27c25ce000-7f27c2782000 r-xp 00000000 00:20 33                         /lib/x86_64-linux-gnu/libc-2.15.so
	7f27c2782000-7f27c2981000 ---p 001b4000 00:20 33                         /lib/x86_64-linux-gnu/libc-2.15.so
	7f27c2981000-7f27c2985000 r--p 001b3000 00:20 33                         /lib/x86_64-linux-gnu/libc-2.15.so
	7f27c2985000-7f27c2987000 rw-p 001b7000 00:20 33                         /lib/x86_64-linux-gnu/libc-2.15.so
	7f27c2987000-7f27c298c000 rw-p 00000000 00:00 0 
	7f27c298c000-7f27c298e000 r-xp 00000000 00:20 31                         /lib/x86_64-linux-gnu/libdl-2.15.so
	7f27c298e000-7f27c2b8e000 ---p 00002000 00:20 31                         /lib/x86_64-linux-gnu/libdl-2.15.so
	7f27c2b8e000-7f27c2b8f000 r--p 00002000 00:20 31                         /lib/x86_64-linux-gnu/libdl-2.15.so
	7f27c2b8f000-7f27c2b90000 rw-p 00003000 00:20 31                         /lib/x86_64-linux-gnu/libdl-2.15.so
	7f27c2b90000-7f27c2ba6000 r-xp 00000000 00:20 95                         /data/jdk1.7.0_67/lib/amd64/jli/libjli.so
	7f27c2ba6000-7f27c2da6000 ---p 00016000 00:20 95                         /data/jdk1.7.0_67/lib/amd64/jli/libjli.so
	7f27c2da6000-7f27c2da7000 rw-p 00016000 00:20 95                         /data/jdk1.7.0_67/lib/amd64/jli/libjli.so
	7f27c2da7000-7f27c2dbf000 r-xp 00000000 00:20 94                         /lib/x86_64-linux-gnu/libpthread-2.15.so
	7f27c2dbf000-7f27c2fbe000 ---p 00018000 00:20 94                         /lib/x86_64-linux-gnu/libpthread-2.15.so
	7f27c2fbe000-7f27c2fbf000 r--p 00017000 00:20 94                         /lib/x86_64-linux-gnu/libpthread-2.15.so
	7f27c2fbf000-7f27c2fc0000 rw-p 00018000 00:20 94                         /lib/x86_64-linux-gnu/libpthread-2.15.so
	7f27c2fc0000-7f27c2fc4000 rw-p 00000000 00:00 0 
	7f27c2fc4000-7f27c2fe6000 r-xp 00000000 00:20 26                         /lib/x86_64-linux-gnu/ld-2.15.so
	7f27c2fe6000-7f27c2ff1000 rw-p 00000000 00:00 0 
	7f27c2ff1000-7f27c300f000 rw-p 00000000 00:00 0 
	7f27c300f000-7f27c3019000 rw-p 00000000 00:00 0 
	7f27c3019000-7f27c30cf000 rw-p 00000000 00:00 0 
	7f27c30cf000-7f27c30d7000 rw-s 00000000 00:20 111                        /tmp/hsperfdata_root/28
	7f27c30d7000-7f27c30da000 ---p 00000000 00:00 0 
	7f27c30da000-7f27c31dc000 rw-p 00000000 00:00 0                          [stack:29]
	7f27c31de000-7f27c31e0000 rw-p 00000000 00:00 0 
	7f27c31e0000-7f27c31e2000 r--s 00000000 00:20 129                        /data/java/test.jar
	7f27c31e2000-7f27c31e3000 rw-p 00000000 00:00 0 
	7f27c31e3000-7f27c31e4000 r--p 00000000 00:00 0 
	7f27c31e4000-7f27c31e6000 rw-p 00000000 00:00 0 
	7f27c31e6000-7f27c31e7000 r--p 00022000 00:20 26                         /lib/x86_64-linux-gnu/ld-2.15.so
	7f27c31e7000-7f27c31e9000 rw-p 00023000 00:20 26                         /lib/x86_64-linux-gnu/ld-2.15.so
	7ffe8021d000-7ffe8023e000 rw-p 00000000 00:00 0                          [stack]
	7ffe80270000-7ffe80272000 r--p 00000000 00:00 0                          [vvar]
	7ffe80272000-7ffe80274000 r-xp 00000000 00:00 0                          [vdso]
	ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
	
	VM Arguments:
	java_command: test.Test1
	Launcher Type: SUN_STANDARD
	
	Environment Variables:
	PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
	
	Signal Handlers:
	SIGSEGV: [libjvm.so+0x99f4a0], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGBUS: [libjvm.so+0x99f4a0], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGFPE: [libjvm.so+0x81a860], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGPIPE: [libjvm.so+0x81a860], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGXFSZ: [libjvm.so+0x81a860], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGILL: [libjvm.so+0x81a860], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGUSR1: SIG_DFL, sa_mask[0]=0x00000000, sa_flags=0x00000000
	SIGUSR2: [libjvm.so+0x81c060], sa_mask[0]=0x00000000, sa_flags=0x10000004
	SIGHUP: [libjvm.so+0x81d160], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGINT: [libjvm.so+0x81d160], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGTERM: [libjvm.so+0x81d160], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	SIGQUIT: [libjvm.so+0x81d160], sa_mask[0]=0x7ffbfeff, sa_flags=0x10000004
	
	
	---------------  S Y S T E M  ---------------
	
	OS:wheezy/sid
	
	uname:Linux 4.1.12-boot2docker #1 SMP Tue Nov 3 06:03:36 UTC 2015 x86_64
	libc:glibc 2.15 NPTL 2.15 
	rlimit: STACK 8192k, CORE 0k, NPROC 1048576, NOFILE 1048576, AS infinity
	load average:0.00 0.74 0.93
	
	/proc/meminfo:
	MemTotal:        4045600 kB
	MemFree:         2462924 kB
	MemAvailable:    3778296 kB
	Buffers:          103732 kB
	Cached:          1211296 kB
	SwapCached:           32 kB
	Active:           412740 kB
	Inactive:         974368 kB
	Active(anon):      81928 kB
	Inactive(anon):    97980 kB
	Active(file):     330812 kB
	Inactive(file):   876388 kB
	Unevictable:           0 kB
	Mlocked:               0 kB
	SwapTotal:       1953572 kB
	SwapFree:        1934472 kB
	Dirty:                80 kB
	Writeback:             0 kB
	AnonPages:         72228 kB
	Mapped:            46096 kB
	Shmem:            107796 kB
	Slab:             157360 kB
	SReclaimable:     137896 kB
	SUnreclaim:        19464 kB
	KernelStack:        2720 kB
	PageTables:         1132 kB
	NFS_Unstable:          0 kB
	Bounce:                0 kB
	WritebackTmp:          0 kB
	CommitLimit:     3976372 kB
	Committed_AS:     555908 kB
	VmallocTotal:   34359738367 kB
	VmallocUsed:       12080 kB
	VmallocChunk:   34359628988 kB
	AnonHugePages:     26624 kB
	HugePages_Total:       0
	HugePages_Free:        0
	HugePages_Rsvd:        0
	HugePages_Surp:        0
	Hugepagesize:       2048 kB
	DirectMap4k:       61376 kB
	DirectMap2M:     4132864 kB
	
	
	CPU:total 4 (4 cores per cpu, 1 threads per core) family 6 model 58 stepping 9, cmov, cx8, fxsr, mmx, sse, sse2, sse3, ssse3, sse4.1, sse4.2, popcnt, avx, aes, tsc, tscinvbit, tscinv
	
	/proc/cpuinfo:
	processor	: 0
	vendor_id	: GenuineIntel
	cpu family	: 6
	model		: 58
	model name	: Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
	stepping	: 9
	microcode	: 0x19
	cpu MHz		: 2594.088
	cache size	: 6144 KB
	physical id	: 0
	siblings	: 4
	core id		: 0
	cpu cores	: 4
	apicid		: 0
	initial apicid	: 0
	fpu		: yes
	fpu_exception	: yes
	cpuid level	: 13
	wp		: yes
	flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 popcnt aes xsave avx rdrand hypervisor lahf_lm
	bugs		:
	bogomips	: 5188.17
	clflush size	: 64
	cache_alignment	: 64
	address sizes	: 36 bits physical, 48 bits virtual
	power management:
	
	processor	: 1
	vendor_id	: GenuineIntel
	cpu family	: 6
	model		: 58
	model name	: Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
	stepping	: 9
	microcode	: 0x19
	cpu MHz		: 2594.088
	cache size	: 6144 KB
	physical id	: 0
	siblings	: 4
	core id		: 1
	cpu cores	: 4
	apicid		: 1
	initial apicid	: 1
	fpu		: yes
	fpu_exception	: yes
	cpuid level	: 13
	wp		: yes
	flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 popcnt aes xsave avx rdrand hypervisor lahf_lm
	bugs		:
	bogomips	: 5188.17
	clflush size	: 64
	cache_alignment	: 64
	address sizes	: 36 bits physical, 48 bits virtual
	power management:
	
	processor	: 2
	vendor_id	: GenuineIntel
	cpu family	: 6
	model		: 58
	model name	: Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
	stepping	: 9
	microcode	: 0x19
	cpu MHz		: 2594.088
	cache size	: 6144 KB
	physical id	: 0
	siblings	: 4
	core id		: 2
	cpu cores	: 4
	apicid		: 2
	initial apicid	: 2
	fpu		: yes
	fpu_exception	: yes
	cpuid level	: 13
	wp		: yes
	flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 popcnt aes xsave avx rdrand hypervisor lahf_lm
	bugs		:
	bogomips	: 5188.17
	clflush size	: 64
	cache_alignment	: 64
	address sizes	: 36 bits physical, 48 bits virtual
	power management:
	
	processor	: 3
	vendor_id	: GenuineIntel
	cpu family	: 6
	model		: 58
	model name	: Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
	stepping	: 9
	microcode	: 0x19
	cpu MHz		: 2594.088
	cache size	: 6144 KB
	physical id	: 0
	siblings	: 4
	core id		: 3
	cpu cores	: 4
	apicid		: 3
	initial apicid	: 3
	fpu		: yes
	fpu_exception	: yes
	cpuid level	: 13
	wp		: yes
	flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 popcnt aes xsave avx rdrand hypervisor lahf_lm
	bugs		:
	bogomips	: 5188.17
	clflush size	: 64
	cache_alignment	: 64
	address sizes	: 36 bits physical, 48 bits virtual
	power management:
	
	
	
	Memory: 4k page, physical 4045600k(2462924k free), swap 1953572k(1934472k free)
	
	vm_info: Java HotSpot(TM) 64-Bit Server VM (24.65-b04) for linux-amd64 JRE (1.7.0_67-b01), built on Jul 25 2014 09:20:21 by "java_re" with gcc 4.3.0 20080428 (Red Hat 4.3.0-8)
	
	time: Thu Jan  7 08:56:15 2016
	elapsed time: 0 seconds

**Comment by irbull** [comment]
Do these crashes happen as soon as you start the system? Or are you calling specific J2V8 API?

**Comment by land007** [comment]
System.load("/data/J2V8/jni/libj2v8_linux_x86_64.so");

**Comment by land007** [comment]
http://54.169.225.61/jni.zip    jni_ubuntu10.04      libj2v8_linux_x86_64.so

**Comment by irbull** [comment]
We are now building against node.js instead of just V8. It is much easier to build against node. I have published build instructions in the README for Mac and Linux. Also, #151 has many more details. I'm closing this as duplicate of #151 now.

**Comment by dzungductran** [comment]
Irbull, Do you have instruction for building Android with Android studio? I built v8 separately on Ubuntu and I have the *.so files.  These are:

out.gn/arm.release/lib.unstripped/libicui18n.cr.so
out.gn/arm.release/lib.unstripped/libv8_libplatform.cr.so
out.gn/arm.release/lib.unstripped/libv8_libbase.cr.so
out.gn/arm.release/lib.unstripped/libicuuc.cr.so
out.gn/arm.release/lib.unstripped/libv8.cr.so
out.gn/arm.release/libicui18n.cr.so
out.gn/arm.release/libv8_libplatform.cr.so
out.gn/arm.release/libv8_libbase.cr.so
out.gn/arm.release/clang_x86_v8_arm/libicui18n.cr.so
out.gn/arm.release/clang_x86_v8_arm/libv8_libplatform.cr.so
out.gn/arm.release/clang_x86_v8_arm/libv8_libbase.cr.so
out.gn/arm.release/clang_x86_v8_arm/libicuuc.cr.so
out.gn/arm.release/libicuuc.cr.so
out.gn/arm.release/libv8.cr.so

What do I do with these files in the Android Studio gradle setup?

Thanks
